File: yt/yt/flow/examples/java/word_count/wordcount/src/test/java/tech/ytsaurus/flow/examples/wordcount/WordCountTestConfiguration.java
package tech.ytsaurus.flow.examples.wordcount;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import tech.ytsaurus.flow.config.CompanionExecutionConfig;
import tech.ytsaurus.flow.context.PipelineContext;
import tech.ytsaurus.flow.execution.CompanionExecution;
import tech.ytsaurus.flow.testutils.CompanionConfigFixtures;
import tech.ytsaurus.flow.testutils.NoServerTestExecution;

@TestConfiguration
public class WordCountTestConfiguration {

    @Bean
    public CompanionExecutionConfig companionExecutionConfig() {
        return CompanionConfigFixtures.defaults();
    }

    @Bean
    public CompanionExecution grpcServerExecution(
            PipelineContext pipelineContext,
            CompanionExecutionConfig companionExecutionConfig
    ) {
        return new NoServerTestExecution(pipelineContext, companionExecutionConfig);
    }
}

File: yt/yt/flow/examples/kotlin/word_count/wordcount/src/test/kotlin/tech/ytsaurus/flow/examples/wordcount/WordCountTestConfiguration.kt
package tech.ytsaurus.flow.examples.wordcount

import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import tech.ytsaurus.flow.config.CompanionExecutionConfig
import tech.ytsaurus.flow.context.PipelineContext
import tech.ytsaurus.flow.execution.CompanionExecution
import tech.ytsaurus.flow.testutils.CompanionConfigFixtures
import tech.ytsaurus.flow.testutils.NoServerTestExecution

@TestConfiguration
open class WordCountTestConfiguration {

    @Bean
    open fun companionExecutionConfig(): CompanionExecutionConfig {
        return CompanionConfigFixtures.defaults()
    }

    @Bean
    open fun grpcServerExecution(
        pipelineContext: PipelineContext,
        companionExecutionConfig: CompanionExecutionConfig,
    ): CompanionExecution {
        return NoServerTestExecution(pipelineContext, companionExecutionConfig)
    }
}
